homeassistant/components/fitbit/sensor.py

```diff
@@ -164,6 +164,15 @@
     }
 )
 
+def get_external_url(hass, require_current_request=False):
+    return get_url(
+        hass,
+        allow_internal=False,
+        allow_ip=False,
+        require_ssl=True,
```

Comment:
This is incorrect. OAuth specification does not require an external URL. Furthermore, because of NAT loopback issues, not everybody can use an external URL, to begin with. Hence, doing that causes different issues.

IP based URLs, Non-SSL & SSL based URLs and internal and external URLs are all valid to use for OAuth authentication.

The current stable version actually looks at the current URL you are using in your browser (and must have been configured in your instance). The dev version diverts a bit.

Reply:
Well, on redirect/callback URL it must be external, no? How the service can access to that otherwise?

Reply:
The service does not access that. With OAuth the applications do not talk directly to each other. Your browser does that via redirects. Hence, it doesn't have to be public (as long as your browser can access it).